util/patch/patch.go

```diff
@@ -291,6 +296,15 @@ func (h *Helper) calculateChanges(after client.Object) (map[string]bool, error)
 		return nil, errors.Wrapf(err, "failed to unmarshal patch data into a map")
 	}
 
+	// log whether a patch is created. If enabled log the full patch.
+	if len(patchDiff) == 0 {
+		log.Info(fmt.Sprintf("No changes for %s", tlog.KObj{Obj: h.after}))
+	} else {
+		if options.LoggingEnabled {
+			log = log.WithValues("patch", sanitizePatch(patchDiff, options.SanitizedLoggingPaths))
```

Comment:
Q: Do we assume that folks want to have the patch explicitly as data / easily parseable? Or would it be more appropriate to add it as part of the message instead of as k/v pair.

Reply:
I like it as part of the k/v pair just because that makes it easier to parse and read using higher level tools.

Reply:
Fine for me